api/v1alpha1/api.go

```diff
@@ -144,3 +166,74 @@ const (
 	// This can be used to describe the routing behavior in HTTPRoute referenced by LLMRoute.
 	LLMModelHeaderKey = "x-envoy-ai-gateway-llm-model"
 )
+
+// LLMProviderAuthenticationType specifies the type of auth mechanism used to access a Provider.
+type LLMProviderAuthenticationType string
+
+const (
+	LLMProviderAuthenticationTypeAPIKey    LLMProviderAuthenticationType = "APIKey"
+	LLMProviderAuthenticationTypeOIDC      LLMProviderAuthenticationType = "OIDC"
+	LLMProviderAuthenticationTypeStaticKey LLMProviderAuthenticationType = "StaticKey"
+)
+
+// +kubebuilder:object:root=true
+
+// BackendSecurityPolicy specifies configuration for authentication and authorization rules on the traffic
+// exiting the gateway to the backend.
+type BackendSecurityPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              BackendSecurityPolicySpec `json:"spec,omitempty"`
+}
+
+// BackendSecurityPolicySpec specifies authentication rules on access the provider from the Gateway.
+type BackendSecurityPolicySpec struct {
+	// Type specifies the auth mechanism used to access the provider. Currently, only "APIKey", AND "OIDC" are supported.
+	//
+	// +kubebuilder:validation:Enum=APIKey;OIDC
+	Type LLMProviderAuthenticationType `json:"type"`
+
+	// BackendRefs are refs of the backends that this BackendSecurityPolicy corresponds to.
+	//
+	// +optional
+	BackendRefs []egv1a1.BackendRef `json:"backendSecurityPolicyRefs,omitempty"`
+
+	// APIKey specific configuration. The API key will be injected into the Authorization header.
+	// +optional
+	APIKey *LLMProviderAPIKey `json:"apiKey,omitempty"`
+
+	// OIDC tokens are retrieved from the following configuration. The token and backend type will determine how Authorization is configured.
+	// +optional
+	OIDC *egv1a1.OIDC `json:"oidc,omitempty"`
```

Comment:
I wonder can we use EG OIDC here?
The EG OIDC need users to provide their identity in the authentication flow, which is not applicable for the authentication at the backend service side.

Reply:
@zhaohuabing you get the JWT authentication token from OIDC provider for exchanging the temporary AWS credential. User needs to provider the `client_id` and `client_secret` to get the OIDC auth token.
https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_oidc.html

Reply:
I think the authentication type here should be AWS IAM, the OIDC token is an intermediate step to get the JWT auth token for exchanging the AWS creds.